src/Qir/Controller/Executable/QuantumExecutableRunner.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Quantum.Qir.Utility;
+using Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
+
+namespace Microsoft.Quantum.Qir.Executable
+{
+    public class QuantumExecutableRunner : IQuantumExecutableRunner
+    {
+        private readonly ILogger logger;
+
+        public QuantumExecutableRunner(ILogger logger)
+        {
+            this.logger = logger;
+        }
+
+        public Task RunExecutableAsync(FileInfo executableFile, EntryPointOperation entryPointOperation, FileInfo outputFile)
```

Comment:
Since the type of linking we use when generating QIR executables is dynamic linking, I think this should receive another argument: `DirectoryInfo librariesDirectory`.

Before running the executable, the content of `librariesDirectory` should be copied into the same directory as the executable. The reason behind this is that it would work in all platforms. Otherwise we would have to edit the `PATH` environment variable or its equivalent depending on the platform.